source/common/upstream/utility.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+#include "envoy/upstream/cluster_manager.h"
+
+namespace Envoy {
+namespace Upstream {
+
+/**
+ * Utility functions for upstream.
+ */
+class Utility {
+public:
+  /**
+   * Resolve the cluster info.
+   * @param decoder_callbacks supplies the decoder callback of filter.
+   * @param cm supplies the cluster manager of the filter.
+   */
+  static Upstream::ClusterInfoConstSharedPtr
+  resolveClusterInfo(Http::StreamDecoderFilterCallbacks* decoder_callbacks,
```

Comment:
I think @fengli79 had issues when putting it in `source/common/http/utility.{cc,h}` due to circular dependencies..